### ISSUE TITLE
Experiment Definition in the DRS

### ISSUE THREAD

**Issue by bnlawrence** [wgcm-wip/drs#3]
We understand that there are a range of ways of using the word experiment, so for the purpose of this ticket, we define experiment as being the aggregation of all properties needed to run a specific simulation, that is
- The era (`CMIP5` or `CMIP6` etc)
- The MIP (`DCMIP`, `DCPP` etc)
- The experiment family (`DCPP-A1`)
- Any experimental ensemble identifier (e.g. start date `1980` for `DCPP-A1`).

How then do we fully define an experiment in the DRS, recognising that many tools will want to want to recognise the notion of experiment as being the above attributes _without_ the ensemble identifier (e.g. `CMIP6/DCPP/DCPP-A1`).

We are agreed that we want a finite number of slots to do this:
- Slot1/ Slot2 / Slot3 / Slot4

Issues we have are:
- What to call "Slot1" (mip-era/activity/project)?,
- What to call "Slot2" (activity/mip)?,
- What to call "Slot3" (experiment-family)?
- What exactly is the fourth slot? A combination of experimental axis identifier (e.g.  perhaps one of start date, driving model, forcing constraint) and axis member identifier (e.g. 1980, model-A, 2XCO2).
- What if we have multiple axes in the fourth slot? (Use a "letter identifier" as we did in the run_variant_id)?
- What if the experiment family has no "internal ensemble"? (Just repeat the family name?)

**Comment by bnlawrence** [comment]
A simpler proposal received via email is to abandon the fourth slot completely, since we really only need it for driving models and DCPP, both of which could be handled in the response ensemble identifier by extending the syntax (to include, say -S1980 or -dI with a look up table for the driving simulation).

**Comment by taylor13** [comment]
In discussing ensembles, I think the first decision is whether to treat something like "S1980" or "src1" (for offline or driving model situations) differently from "r1i3p22f3".  (Later we can decide how to package the information in the DRS).  I think we should treat these ensembles differently.  Here are things that distinguish type 1 from type 2 ensembles (in both cases the ensemble should be a group of experiments that are quite similar except for one thing (e.g., start date, forcing, driving-source):

Ensemble type 1:  
      *called for by experiment design with a limited number of permitted members (e.g., only start dates 1950-2016 are called for) 
      \* the identifying label is meaningful and has the same meaning in all models 
      \* known examples:  DCCP experiments with different start dates;  offline ice sheet models driven by one of the CMIP6 global coupled model simulations; single forcing experiments called for by DAMIP, e.g., histVLC, histSOZ, histGHG  (also offline readiation and land-surface simulations?)

Ensemble type 2:
       \* although permitted by the experiment design, there are generally unlimited options for creating new ensemble members (e.g., for perturbed physics, parameter values can be varied continuously; for "realizations", a nearly unlimited number of possible initializations times (from different points in the control)
      \*  the same identifying label (e.g., an index) may have different meanings, depending on the model
      \*  known examples: "realization", "initialization" procedure (only used for DCPP runs), perturbed "physics" (might be needed by a handful of models or fewer), perturbed "forcing" (used to explore a non-standard forcing data set).  
      \* the only known type 2 ensemble required is really "realization".  All the others are included to allow modeling groups to carry out sensitivity studies permitted within the defined experiment, but not explicitly included.

**Comment by MartinaSt** [comment]
I miss the "model" or source_id in your 4 slots definition. Am I right that this is "Slot2", which includes institute and model/source_id (mip/institute/model)? And the citation granularities are thus "Slot2" and "Slot3"?

The definitions of "Slot1" and "Slot2" both include activity? That is a bit confusing for me. I would call "Slot2": mip/model.

**Comment by taylor13** [comment]
"source" is not needed to identify the experiment (which is defined solely by the conditions specified for making a run.

**Comment by bnlawrence** [comment]
@MartinaSt Yes, @taylor13 is right. The model/source is part of the simulation identifier.  However, I think your underlying issue is you need to aggregate on the model. Over the weekend I intend to write a short document discriminating between the need to use the DRS as a tool for identifying atomic datasets (and thus for faceted browse) and using DRS terms to create identifiers for aggregations of atomic datasets of interest (including those which are associated with external information collections such as the ones we need for es-doc and citation).

**Comment by martinjuckes** [comment]
@bnlawrence sounds like a useful document. The relationship between the metadata used for search and discovery and the metadata used for organising the data should be as direct and transparent as possible, but it is also important to understand the differences (especially with regard to redundancy, which, I believe, is useful in search and a nuisance in data management).
